local/rest_api_gcbm/gcbm.py

```diff
@@ -254,40 +323,62 @@ def add_file(self, file_path: str, category: str = ""):
+            if self.db._append(filename, check):
+                # This just stores, and doesn't write anything yet...
+                self.db._update_config()
 
         if category != "":
-            if category == "disturbances":
+            if category == "disturbance":
                 _disturbance(filename, check=False)
-            elif category == "classifiers":
+            elif category == "classifier":
+                print("Found it here: ", filename)
                 _classifier(filename, check=False)
             elif category == "miscellaneous":
                 _miscellaneous(filename, check=False)
+            elif category == "db":
+                _db(filename, check=False)
             else:
-                # We can also raise an error here
-                raise UserWarning(f"Given category {category} not supported, supported categories are: {disturbances, classifiers, miscellaneous}")
+                raise ValueError(
+                    f"Given category {category} not supported, supported categories are: {self.supported_categories}"
+                )
         else:
-            print(f"Category wasn't provided, attempting to deduce it from the file path: {file_path}")
-            if self.disturbances._append(filename):
-                self.disturbances._update_config()
-                return
-            if self.classifiers._append(filename):
-                self.classifiers._update_config()
-                return
-            if self.miscellaneous._append(filename):
-                self.miscellaneous._update_config()
-                return 
-            print(f"Couldn't deduce a valid from the file path {file_path}, supported categories: {self.supported_categories}")
+            deduced_category = ""
+            if self.disturbance._append(filename):
+                self.json_paths.update(self.disturbance._update_config())
+                deduced_category = "disturbance"
+            elif self.classifier._append(filename):
+                self.json_paths.update(self.classifier._update_config())
+                deduced_category = "classifier"
+            elif self.miscellaneous._append(filename):
+                self.json_paths.update(self.miscellaneous._update_config())
+                deduced_category = "miscellaneous"
+            elif self.db._append(filename):
+                self.db._update_config()
+                deduced_category = "db"
+            else:
+                raise ValueError(
+                    f"Couldn't deduce a valid from the file path {file_path}, supported categories: {self.supported_categories}"
+                )
+            print(
+                    f"Category wasn't provided, deduced category: {deduced_category} it from the file path: {file_path}"
+            )
 
+    def generate_configs(self):
+        self._generate_provider_config(self.dirpath)
+        self._generate_modules_cbm_config(self.dirpath)
+        self._add_variables(self.dirpath)
```

Comment:
This function should be called if someone wants to update the provider/modules_cbm/variables files. I cross-checked that all the files are generated correctly, but I'll appreciate another set of eyes.

TODO for me: write tests to cross-check this :)